compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/java/KSFunctionDeclarationJavaImpl.kt

```diff
@@ -81,15 +86,29 @@ class KSFunctionDeclarationJavaImpl private constructor(val psi: PsiMethod) : KS
     }
 
     override val returnType: KSTypeReference? by lazy {
-        if (psi.returnType != null) {
-            KSTypeReferenceJavaImpl.getCached(psi.returnType!!)
-        } else {
-            null
+        when {
+            psi.returnType != null -> {
+                KSTypeReferenceJavaImpl.getCached(psi.returnType!!)
+            }
+            psi.isConstructor -> {
+                psi.containingClass?.let { containingClass ->
+                    KSTypeReferenceLiteJavaImpl.getCached(
+                        KSClassDeclarationJavaImpl.getCached(containingClass).asType(emptyList())
```

Comment:
`asType(emptyList())` is undocumented and leaks default type, which is also undocumented. We should probably block it (by checking the size) and encourage users to use `startProejctedType` when type arguments are irrelevant.

Reply:
is there a different function i should use here?
I'm trying to get a type refernce to the PsiClass cheapest way possible.

Reply:
Please try `asStarProjectedType()`, which is slightly more expensive than `asType(emptyList()` but well defined.

Reply:
thanks.